src/base/RandomUuid.cc

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 1996-2022 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+
+#include "base/IoManip.h"
+#include "base/RandomUuid.h"
+#include "base/TextException.h"
+#include "defines.h"
+
+#include <iostream>
+#include <random>
+
+static_assert(sizeof(RandomUuid) == 128/8, "RandomUuid has RFC 4122-prescribed 128-bit size");
+
+RandomUuid::RandomUuid()
+{
+    // Generate random bits for populating our UUID.
+    // STL implementation bugs notwithstanding (e.g., MinGW bug #338), this is
+    // our best change of getting a non-deterministic seed value for the r.n.g.
+    static std::random_device dev; // unknown a priori size (sizeof(int)) values
+    static std::mt19937_64 rng(dev()); // known 64-bit size values
```

Comment:
Done.

Reply:
> 5d15b3a9c496cafcbbb0f628852fd31107a079c1: We cannot get rid of a local random_device variable because it is passed by reference to mt19937_64 constructor.

We do not pass `dev` to the constructor. What we pass is the _result_ of `dev.operator()` call, which is just an integer. Try something like the following, perhaps?

```suggestion
    static std::mt19937_64 rng(std::random_device()); // produces 64-bit sized values
```

Reply:
We can even make this functionality reusable (e.g., in the future random ACL refactoring) and the code more readable:

```C++
/// A non-deterministic integer for priming a deterministic random number generator.
/// This function may be very slow. Avoid calling it at transaction processing time!
inline
std::random_device::result_type
RngSeed()
{
    // STL implementation bugs notwithstanding (e.g., MinGW bug #338), this is
    // our best change of getting a non-deterministic seed value for the r.n.g.
    std::random_device device; // not "static" to avoid locking entropy sources
    return device();
}
```

The above snippet could go into existing src/SquidMath.h but, due to the required `#include`s, it is better to place it into a new header file (e.g., src/base/Random.h). However, this is more out-of-scope work, and I do not insist on any of it. It can be viewed as the responsibility of whoever adds the _second_ piece of code that needs such a r.n.g. seed.

Reply:
> We do not pass dev to the constructor.

Got it, I misread the documentation. I decided to follow the simpler approach (f021669) - we can add RngSeed() later.

Reply:
Thank you!

> https://github.com/squid-cache/squid/commit/f02166991762b0aa35234b22860ccff4fcfbd19a: it looks better to construct it each time instead of keeping it in a static variable

It will be constructed and used exactly once in both implementations. The difference is the "keeping" part.